### DISCUSSION TITLE
Alternative of tox w.r.t CUE tests

### DISCUSSION THREAD

**Discussion by debuggerrr** [cue-lang/cue#2227] [General]
I was wondering if there could be anything like `tox in python` as standard testing practice. I did research but didn't find anything. Can anyone help me with how can I achieve similar things in CUE?

**Comment by debuggerrr** [comment]
@mvdan Can you help here?

**Reply by verdverm** [reply]
Can you expand on what you are trying to do? Do you want to test CUE code or have a testing framework based on CUE?

Providing as much context and information as possible will help us answer your questions, generally speaking

**Reply by debuggerrr** [reply]
@verdverm  I am trying to test the negative scenarios but I want my tests to pass. So, something like if my code is going to raise an error, it should match with the exception mentioned in the core logic like `self.assertRaises`. Also, I am looking for something like [tox](https://tox.wiki/en/latest/) which can be environment agnostic that can be integrated with the python project where I can keep these CUE files (data and test both). I hope I am much clear now.
